### PR TITLE
gnrc_ipv6_nib: don't add non-link-local to NC without ARSM 

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
This fixes https://github.com/RIOT-OS/RIOT/issues/9709. When there is no address resolution state-machine (ARSM), address resolution is restricted to reverse translations of the interface identifier (IID) of link-local addresses as defined in [RFC 6775](https://tools.ietf.org/html/rfc6775#section-5.6). As such, the function to fetch the next-hop data from the NIB-internals, asserts that the address of the next-hop is a link-local address, when the ARSM is deactivated.

https://github.com/RIOT-OS/RIOT/blob/d2369a2651400cd7f1ae1011332210654fdc4f04/sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c#L333-L338

However, there are functions that add to the neighbor cache without checking if the address is a link-local address or not. This PR aims to fix that.

### Testing procedure
The steps to reproduce in https://github.com/RIOT-OS/RIOT/issues/9709#issuecomment-653298937 should succeed now and also the one described in OP.

`examples/gnrc_networking` should still work on `native` when using global addresses:

<details>

```
$ make -C examples/gnrc_networking -j flash term
[…]
main(): This is RIOT! (Version: 2020.10-devel-1136-gc5b63)
RIOT network stack example application
All up, running the shell now
> ifconfig 6 add affe::2
ifconfig 6 add affe::2
success: added affe::2/64 to interface 6
```

```
$ PORT=tap1 make -C examples/gnrc_networking term
[…]
main(): This is RIOT! (Version: 2020.10-devel-1136-gc5b63)
RIOT network stack example application
All up, running the shell now
> ifconfig
ifconfig
Iface  6  HWaddr: E2:BC:7D:CB:F5:50 
          L2-PDU:1500  MTU:1500  HL:64  RTR  
          RTR_ADV  
          Source address length: 6
          Link type: wired
          inet6 addr: fe80::e0bc:7dff:fecb:f550  scope: link  VAL
          inet6 addr: affe::e0bc:7dff:fecb:f550  scope: global  VAL
          inet6 group: ff02::2
          inet6 group: ff02::1
          inet6 group: ff02::1:ffcb:f550
          
          Statistics for Layer 2
            RX packets 90  bytes 8276
            TX packets 64 (Multicast: 13)  bytes 5824
            TX succeeded 64 errors 0
          Statistics for IPv6
            RX packets 90  bytes 7016
            TX packets 64 (Multicast: 13)  bytes 4928
            TX succeeded 64 errors 0

> ping6 affe::2
ping6 affe::2
12 bytes from affe::2: icmp_seq=0 ttl=64 time=3.056 ms
12 bytes from affe::2: icmp_seq=1 ttl=64 time=1.293 ms
12 bytes from affe::2: icmp_seq=2 ttl=64 time=1.294 ms

--- affe::2 PING statistics ---
3 packets transmitted, 3 packets received, 0% packet loss
round-trip min/avg/max = 1.293/1.881/3.056 ms
```

</details>

### Issues/PRs references
Fixes https://github.com/RIOT-OS/RIOT/issues/9709

### CODE REVIEW

File: sys/net/gnrc/network_layer/ipv6/nib/nib.c [hunk 1]
```diff
@@ -237,7 +237,11 @@ int gnrc_ipv6_nib_get_next_hop_l2addr(const ipv6_addr_t *dst,
                       "search neighbor cache\n",
                       ipv6_addr_to_str(addr_str, dst, sizeof(addr_str)));
 
-                if (res == 0) {
+                if ((res == 0) &&
+                    /* If ARSM is not active don't use non-link-local as
+                     * next hop */
+                    (IS_ACTIVE(CONFIG_GNRC_IPV6_NIB_ARSM) ||
+                     ipv6_addr_is_link_local(dst))) {
```
Comment (benpicco) [inline sys/net/gnrc/network_layer/ipv6/nib/nib.c:244]: The comment doesn't match the expression

Comment (miri64) [reply sys/net/gnrc/network_layer/ipv6/nib/nib.c:244]: Yes it does, but I grant that it is confusing with the double negative. Will fix ;-)

**Comment by miri64** [comment]
@benpicco told me to squash offline ;-).

**Review by pokgak — APPROVED** [review]
Confirmed this fixes #9709.

dtls-echo example without this PR (master 0aca11696dcf173a40a45835b1f9624c34bf330d):
```
ifconfig 6 add 2001:db8::2
2020-09-01 13:47:55,042 # ifconfig 6 add 2001:db8::2
2020-09-01 13:47:55,046 # success: added 2001:db8::2/64 to interface 6
dtlsc 2001:db8::1 "HELLO WORLD"
2020-09-01 13:47:58,384 #  dtlsc 2001:db8::1 "HELLO WORLD"
2020-09-01 13:48:08,384 # Failed to send the data
2020-09-01 13:48:08,385 # 0x38bb
2020-09-01 13:48:08,387 # *** RIOT kernel panic:
2020-09-01 13:48:08,389 # FAILED ASSERTION.
2020-09-01 13:48:08,389 # 
2020-09-01 13:48:08,390 # *** halted.
2020-09-01 13:48:08,390 # 
```

dtls-echo example with this PR:
```
ifconfig 6 add 2001:db8::2
2020-09-01 13:45:13,165 #  ifconfig 6 add 2001:db8::2
2020-09-01 13:45:13,169 # success: added 2001:db8::2/64 to interface 6
dtlsc 2001:db8::1 "HELLO WORLD"
2020-09-01 13:45:23,562 #  dtlsc 2001:db8::1 "HELLO WORLD"
2020-09-01 13:45:33,563 # Failed to send the data
```

**Review by benpicco — APPROVED** [review]
Code makes sense too.

**Pull request merged**